src/compiler/parser-quirk-vafirst.bas

```diff
@@ -159,9 +151,20 @@ private sub hSolveValistType _
 			n->dtype = typeJoinDtOnly( typeDeref( n->dtype ) , FB_DATATYPE_VA_LIST )
 
 		case FB_CVA_LIST_BUILTIN_C_STD
-			'' va_list[1] gets passed as a pointer
-			'' *cast( __builtin_va_list ptr, @expr )
-			n = astNewDeref( n, typeAddrOf( typeJoinDtOnly( n->dtype, FB_DATATYPE_VA_LIST ) ), astGetSubType( n ) )
+			if( astIsVAR( n ) ) then
+				if( symbIsParamByval( astGetSymbol( n ) ) ) then
+					exit select
```

Comment:
looks like `exit select` in the wrong place

Reply:
The first exit select is actually correct.  The code immediately following was something I was trying to do but it didn't work:  Here's what it should look like, with comments:
```
case FB_CVA_LIST_BUILTIN_C_STD
	if( astIsVAR( n ) ) then
		if( symbIsParamByval( astGetSymbol( n ) ) ) then

			'' The dtype & subtype tell us that this is a
			'' C struct array.  Only the pointer to the
			'' array is acutally passed by value.  Taking
			'' the address of this symbol actually gives
			'' us the address of the passed-by-value-pointer
			'' to the the array instead and we don't want that.
			'' We also can't, deref the symbol directly, 
			'' because C won't allow deref on an array.
			''					
			'' Give up and let gcc backend use the param var
			'' symbol name as-is.  The mangle modifier will
			'' let C backend know it's the va_list type, 
			'' and exprNewVREG() won't try to deref it.

			exit select
		end if
	end if

	'' for anything else, replace the dtype here and let backend
	'' handle it (though there are some casts in exprNewVREG that
	'' could be solved out, TODO).  Taking the address of array
	'' variable (in C) is the same address as just referencing
	'' the array name.  e.g. "int a[10]", a = &a = &(a[0]), but
	'' different pointer types

	'' cast( __builtin_va_list, expr )
	n->dtype = typeJoinDtOnly( n->dtype, FB_DATATYPE_VA_LIST )
```
In theory should work for any variable and skip any casting, not just byval param variables.